lib/_stream_readable.js

```diff
@@ -437,7 +441,8 @@ Readable.prototype.read = function(n) {
     if (state.length === 0)
       state.needReadable = true;
     // call internal read method
-    this._read(state.highWaterMark);
+    if (!state.destroyed)
```

Comment:
I think this check can me moved in the previous `if` block: `state.ended || state.reading || state.destroyed`.